pkg/dns/provider/errors/handlererror.go

```diff
@@ -20,12 +20,26 @@ import (
 	"fmt"
 )
 
+type wrappedError struct {
+	msg string
+	err *handlerError
+}
+
+func (e *wrappedError) Error() string {
+	return fmt.Sprintf("%s: %s", e.msg, e.err.Error())
+}
+
+func (e *wrappedError) Cause() error {
+	return e.err
+}
+
 type handlerError struct {
+	msg string
 	err error
 }
 
 func WrapAsHandlerError(err error, msg string) error {
-	return fmt.Errorf("%s: %w", msg, &handlerError{err: err})
+	return &wrappedError{msg: msg, err: &handlerError{err: err}}
```

Comment:
Why this, I think %w does exactly this kind of wrapping

Reply:
Changed handling to deal with `Unwrap` method